client/configuration/search_path.py

```diff
@@ -72,15 +81,55 @@ class SitePackageElement(Element):
     package_name: str
     is_toplevel_module: bool = False
 
-    def package_path(self) -> str:
-        module_suffix = ".py" if self.is_toplevel_module else ""
-        return self.package_name + module_suffix
+    def package_path(self) -> Union[str, None]:
+        if not self.is_toplevel_module:
+            return self.package_name
 
-    def path(self) -> str:
-        return os.path.join(self.site_root, self.package_path())
+        this_pkg_filter = re.compile(
+            r"{}-([0-99]\.)*dist-info(/)*.*".format(self.package_name)
+        )
 
-    def command_line_argument(self) -> str:
-        return self.site_root + "$" + self.package_path()
+        if self.site_root not in dist_info_in_root:
+            dist_info_in_root[self.site_root] = []
+            for directory in os.listdir(self.site_root):
+                if _site_filter.fullmatch(directory) is not None:
+                    dist_info_in_root[self.site_root].append(directory)
+
+        for directory in dist_info_in_root[self.site_root]:
+            if this_pkg_filter.fullmatch(directory):
+                dist_info_path = f"{self.site_root}/{directory}"
+                break
+        else:
+            return None
+
+        not_toplevel_patterns: Tuple[re.Pattern[str], re.Pattern[str]] = (
+            this_pkg_filter,
+            _PYCACHE,
+        )
+
+        # pyre-fixme[61]: Local variable `dist_info_path` is undefined, or not always defined.
+        with open(file=f"{dist_info_path}/RECORD", mode="r") as record:
```

Comment:
could you make sure this is always defined?

Reply:
It is a known issue that pyre doesn't understand for-else. #791

Reply:
hmm yeah if that's the case it would be better if you could factor out the for else loop to please the static analysis

Reply:
I don't think so.I prefer to leave this fixme there until #791 fixed since it is a bug of Pyre.